src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -894,118 +925,344 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
+            //•	the safe-to-escape of all argument expressions(including the receiver)
+            //
+            // Note that these rules are identical to the above rules for ref-safe - to - escape, but apply only when the return type is a ref struct type.
+
+            //by default it is safe to escape
+            uint escapeScope = Binder.ExternalScope;
+
+            ArrayBuilder<bool> inParametersMatchedWithArgs = null;
+
+            if (!args.IsDefault)
             {
-                var indexerAccess = expr as BoundIndexerAccess;
-                if (indexerAccess != null)
+                for (var argIndex = 0; argIndex < args.Length; argIndex++)
                 {
-                    var indexer = indexerAccess.Indexer;
+                    RefKind effectiveRefKind = GetEffectiveRefKind(argIndex, argRefKinds, parameters, argsToParamsOpt, ref inParametersMatchedWithArgs);
 
-                    return CheckArgumentsReturnable(
-                        propertySyntax,
-                        indexer,
-                        indexer.Parameters,
-                        indexerAccess.Arguments,
-                        indexerAccess.ArgumentRefKindsOpt,
-                        indexerAccess.ArgsToParamsOpt,
-                        checkingReceiver,
-                        diagnostics);
+                    // if ref is passed and it is not a ref-like, check ref escape, 
+                    // since it is the same or narrower than val escape and callee may wrap the ref into a val
+                    var argument = args[argIndex];
+                    var argEscape = effectiveRefKind != RefKind.None && argument.Type.IsByRefLikeType == false ?
+                                        GetRefEscape(args[argIndex], scopeOfTheContainingExpression) :
+                                        GetValEscape(args[argIndex], scopeOfTheContainingExpression);
+
+                    escapeScope = Math.Max(escapeScope, argEscape);
+
+                    if (escapeScope >= scopeOfTheContainingExpression)
+                    {
+                        // no longer needed
+                        inParametersMatchedWithArgs?.Free();
+
+                        // can't get any worse
+                        return escapeScope;
+                    }
                 }
             }
 
-            return true;
+            // handle omitted optional "in" parameters if there are any
+            ParameterSymbol unmatchedInParameter = TryGetUnmatchedInParameterAndFreeMatchedArgs(parameters, ref inParametersMatchedWithArgs);
+
+            // unmatched "in" parameter is same as a literal, its ref escape is scopeOfTheContainingExpression  (can't get any worse)
+            //                                                its val escape is ExternalScope                   (does not affect overal result)
+            if (unmatchedInParameter != null)
+            {
+                return scopeOfTheContainingExpression;
+            }
+
```

Comment:
Extra blank line

Reply:
Just one line separating two `if`s . Or is this about something else?

---
In reply to: [135156229](https://github.com/dotnet/roslyn/pull/21334#discussion_r135156229) [](ancestors = 135156229)